File: data-structures/graphs/PHP/Graph.php
<?php

// Graph represents a directed graph using list
class Graph
{
    public $vertices = [];

    public function addVertex($vertex)
    {
        $this->vertices[] = $vertex;
    }

    public function addEdge($vertex1, $vertex2)
    {
        $vertex1->edges[] = $vertex2;
    }

    public function print()
    {
        $string = '';
        foreach ($this->vertices as $vertex) {
            $string .= "\n" . $vertex->data . ' -> ';
            foreach ($vertex->edges as $edge) {
                $string .= $edge->data . " ";
            }
        }
        echo $string;
    }
}

class Vertex
{
    public $data;
    public $edges = [];

    public function __construct($data)
    {
        $this->data = $data;
    }
}

$graph = new Graph();

$vertexA = new Vertex("A");
$vertexB = new Vertex("B");
$vertexC = new Vertex("C");

$graph->addVertex($vertexA);
$graph->addVertex($vertexB);
$graph->addVertex($vertexC);

$graph->addEdge($vertexA, $vertexB);
$graph->addEdge($vertexB, $vertexC);
$graph->addEdge($vertexC, $vertexB);
$graph->addEdge($vertexC, $vertexA);
$graph->addEdge($vertexA, $vertexC);

$graph->print();

File: algorithms/graphs/breadth-first-search/Go/graph.go
package main

import "fmt"

// Graph represents a directed graph using an adjency list
type Graph struct {
	vertices map[int]*Vertex
}

// Vertex represents a vertex in the graph
type Vertex struct {
	key   int
	edges map[int]*Vertex
}

// AddVertex adds a new vertex in the graph
func (g *Graph) AddVertex(key int) {
	g.vertices[key] = newVertex(key)
}

// AddEdge add a new edge between two vertices
func (g *Graph) AddEdge(from, to int) {
	g.vertices[from].edges[to] = g.vertices[to]
}

// GetVertex returns a vertex by key
func (g *Graph) GetVertex(search int) *Vertex {
	for key, vertex := range g.vertices {
		if key == search {
			return vertex
		}
	}
	return nil
}

// Print displays the content of the graph
func (g *Graph) Print() {
	for key, vertex := range g.vertices {
		fmt.Println()
		fmt.Printf("%v -> ", key)
		edges := vertex.edges
		for _, edge := range edges {
			fmt.Printf("%v ", edge.key)
		}
	}
	fmt.Println()
}

// NewGraph creates a new Graph
func NewGraph() Graph {
	return Graph{vertices: make(map[int]*Vertex)}
}

func newVertex(key int) *Vertex {
	return &Vertex{key: key, edges: make(map[int]*Vertex)}
}
